# Kernel SVM

## Importing the libraries

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Importing the dataset

In [2]:
dataset = pd.read_csv('500_Person_Gender_Height_Weight_Index.csv')
X = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1].values

## Splitting the dataset into the Training set and Test set

In [3]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2)

## Feature Scaling

In [4]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

## Training the Kernel SVM model on the Training set

In [5]:
from sklearn.svm import SVC
classifier = SVC(kernel = 'rbf', random_state = 0)
classifier.fit(X_train, y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='rbf',
    max_iter=-1, probability=False, random_state=0, shrinking=True, tol=0.001,
    verbose=False)

## Predicting a new result

In [6]:
print(classifier.predict(sc.transform([[30,87000]])))

[4]


## Predicting the Test set results

In [7]:
y_pred = classifier.predict(X_test)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

[[4 4]
 [5 5]
 [5 5]
 [5 5]
 [2 2]
 [5 5]
 [5 5]
 [5 5]
 [2 2]
 [4 4]
 [3 3]
 [4 4]
 [5 5]
 [4 5]
 [3 3]
 [4 4]
 [4 4]
 [3 3]
 [2 2]
 [5 5]
 [3 3]
 [5 4]
 [5 5]
 [5 5]
 [3 3]
 [5 5]
 [4 4]
 [2 2]
 [5 5]
 [5 5]
 [1 0]
 [3 3]
 [3 3]
 [5 5]
 [4 4]
 [4 4]
 [4 4]
 [4 4]
 [2 2]
 [1 2]
 [5 5]
 [1 1]
 [5 5]
 [2 1]
 [5 5]
 [2 2]
 [4 4]
 [2 2]
 [5 5]
 [5 5]
 [5 5]
 [5 5]
 [5 5]
 [5 5]
 [1 0]
 [3 3]
 [4 4]
 [4 4]
 [4 4]
 [5 4]
 [5 5]
 [4 4]
 [2 2]
 [5 5]
 [5 5]
 [2 2]
 [5 5]
 [4 4]
 [2 2]
 [1 2]
 [3 4]
 [4 4]
 [4 4]
 [4 4]
 [5 5]
 [3 3]
 [5 5]
 [5 5]
 [5 5]
 [5 5]
 [4 4]
 [5 4]
 [5 5]
 [2 2]
 [4 4]
 [5 5]
 [5 5]
 [3 3]
 [5 5]
 [4 4]
 [1 1]
 [2 2]
 [5 5]
 [5 5]
 [5 5]
 [5 5]
 [4 4]
 [5 5]
 [5 5]
 [5 4]]


## Making the Confusion Matrix

In [8]:
from sklearn.metrics import confusion_matrix, accuracy_score
cm = confusion_matrix(y_test, y_pred)
print(cm)
accuracy_score(y_test, y_pred)

[[ 0  2  0  0  0  0]
 [ 0  2  1  0  0  0]
 [ 0  2 12  0  0  0]
 [ 0  0  0 10  0  0]
 [ 0  0  0  1 23  4]
 [ 0  0  0  0  1 42]]


0.89

# Conclusion
> In this model I have used data of 500 person Gender, Height, Weight and Index

> data terms:

1. Gender : Male / Female
2. Height: Number (cm)
3. Weight: Number (Kg)
4. Index :
  * 0 - Extremely Weak
  * 1 - Weak
  * 2 - Normal
  * 3 - Overweight
  * 4 - Obesity
  * 5 - Extreme Obesity

> Conclusion:
Support Vector Machine (SVM) Classifier is one of the perfect model to predict the index based on the gender's height and weight
<br>
As model gives the 89% accuracy